File: adapter/oauth/src/main/java/com.hearlers.gateway/OAuthProviderClient.java
package com.hearlers.gateway;


import com.hearlers.gateway.auth.model.AuthInfo;
import com.hearlers.gateway.command.AuthCommand;

public interface OAuthProviderClient {
    AuthInfo.TokenInfo getToken(String code, String state, String clientId);
    AuthInfo.OAuthUserInfo getOAuthUser(String accessToken);
}

File: adapter/oauth/src/main/kotlin/com/hearlers/gateway/OAuthProviderClient.kt
package com.hearlers.gateway

import com.hearlers.gateway.auth.model.AuthInfo.OAuthUserInfo
import com.hearlers.gateway.auth.model.AuthInfo.TokenInfo

interface OAuthProviderClient {
    fun getToken(code: String, state: String, clientId: String): TokenInfo
    fun getOAuthUser(accessToken: String): OAuthUserInfo
}